src/features/defi/providers/cosmos/components/CosmosManager/Withdraw/components/Confirm.tsx

```diff
@@ -0,0 +1,194 @@
+import { Alert, AlertIcon, Box, Stack } from '@chakra-ui/react'
+import { toAssetId } from '@shapeshiftoss/caip'
+import { Confirm as ReusableConfirm } from 'features/defi/components/Confirm/Confirm'
+import { Summary } from 'features/defi/components/Summary'
+import {
+  DefiParams,
+  DefiQueryParams,
+  DefiStep,
+} from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import {
+  assetIdToUnbondingDays,
+  StakingAction,
+} from 'plugins/cosmos/components/modals/Staking/StakingCommon'
+import { useStakingAction } from 'plugins/cosmos/hooks/useStakingAction/useStakingAction'
+import { getStakingFees } from 'plugins/cosmos/utils'
+import { useContext, useEffect, useMemo, useState } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { Amount } from 'components/Amount/Amount'
+import { AssetIcon } from 'components/AssetIcon'
+import { StepComponentProps } from 'components/DeFi/components/Steps'
+import { Row } from 'components/Row/Row'
+import { RawText, Text } from 'components/Text'
+import { useBrowserRouter } from 'hooks/useBrowserRouter/useBrowserRouter'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import {
+  selectAssetById,
+  selectMarketDataById,
+  selectPortfolioCryptoHumanBalanceByAssetId,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { CosmosWithdrawActionType } from '../WithdrawCommon'
+import { WithdrawContext } from '../WithdrawContext'
+
+export const Confirm = ({ onNext }: StepComponentProps) => {
+  const [gasLimit, setGasLimit] = useState<string | null>(null)
+  const [gasPrice, setGasPrice] = useState<string | null>(null)
+  const { state, dispatch } = useContext(WithdrawContext)
+  const translate = useTranslate()
+  const { query } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { chainId, contractAddress, assetReference } = query
+
+  const assetNamespace = 'slip44' // TODO: add to query, why do we hardcode this?
```

Comment:
Will tackle in a separate PR, this will have to be done consistently across all DeFi opportunities, this is copied and paste from the current implementations